### ISSUE TITLE
SA1021 Code fix incorrect & possible false positive

### ISSUE THREAD

**Issue by elachlan** [DotNetAnalyzers/StyleCopAnalyzers#3428]
Example:
https://github.com/dotnet/msbuild/blob/7852153acab419ef11b98a3868bb9751130f40b9/src/Shared/CommunicationsUtilities.cs#L591-L594
`Trace(/* nodeId */ -1, format, args);`

Tries to fix it to:
`Trace(-/* nodeId */1, format, args);`

The negative sign is being moved to the start of the section and not taking into account comments.

Second Example:
https://github.com/dotnet/msbuild/blob/7852153acab419ef11b98a3868bb9751130f40b9/src/Build.UnitTests/Evaluation/Expander_Tests.cs#L3455-L3457
`Assert.Equal((~-43).ToString(), result);`

'Fixes' it to:
`Assert.Equal((~ -43).ToString(), result);`

Which then results in IDE0055 suggested fix. I think this is a false positive. But I am not 100% on what should be done.

**Comment by sharwell** [comment]
📝 The expected (manual) solution here is the following:

```csharp
Trace(nodeId: -1, format, args);
```
